---
title: SPAdes
type: docs
toc: false
---




TODO:

A. Background and Development
   1. Origin and creators
      - Developed at St. Petersburg Academic University
      - Led by Pavel Pevzner's team
   2. Initial release and purpose
      - First released in 2012
      - Designed to address challenges in single-cell and short-read sequencing
B. Key Features and Innovations
   1. Versatility
      - Suitable for both single-cell and standard (multicell) assembly
      - Adaptable to various sequencing technologies
   2. Core algorithmic innovations
      - Multisized de Bruijn graph approach
      - Advanced error correction tailored for different data types
      - Paired assembly graph for improved use of read-pair information
   3. Specialized capabilities
      - Effective handling of uneven coverage (crucial for single-cell data)
      - Hybrid assembly combining short and long reads
      - Dedicated modes for metagenome, transcriptome, and plasmid assembly
C. Impact and Adoption
   1. Widespread use in genomics research
   2. Continuous development and community support


<!-- REFERENCES -->

[^bankevich2012spades]: Bankevich, A., Nurk, S., Antipov, D., Gurevich, A. A., Dvorkin, M., Kulikov, A. S., ... & Pevzner, P. A. (2012). SPAdes: a new genome assembly algorithm and its applications to single-cell sequencing. Journal of computational biology, 19(5), 455-477. DOI: [10.1089/cmb.2012.0021](https://doi.org/10.1089/cmb.2012.0021)
[^prjibelski2014exspander]: Prjibelski, A. D., Vasilinetc, I., Bankevich, A., Gurevich, A., Krivosheeva, T., Nurk, S., ... & Pevzner, P. A. (2014). ExSPAnder: a universal repeat resolver for DNA fragment assembly. Bioinformatics, 30(12), i293-i301. DOI: [10.1093/bioinformatics/btu266](https://doi.org/10.1093/bioinformatics/btu266)
